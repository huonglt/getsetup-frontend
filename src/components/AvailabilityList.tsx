import { Typography, Box } from '@mui/material'
import React from 'react'
import { GuideAvailability } from '../types/guide'
import { formatAvailabilityTime } from '../util/util'

type Props = {
  guideAvailability: GuideAvailability
}

/**
 * UI to show guide teaching availability in list format
 * Will improve to render in calendar style format in the future
 */
export const AvailabilityList = (props: Props) => {
  const { userId, weekNumber, availability } = props.guideAvailability

  return (
    <Box className="spaceTop">
      <Typography>
        Teaching Availability for userId {userId} for the week #{weekNumber}
      </Typography>
      {availability
        .filter((timeslot) => timeslot.from && timeslot.to)
        .map((timeslot, index) => {
          return (
            <Typography key={index}>
              {formatAvailabilityTime(timeslot)}
            </Typography>
          )
        })}
    </Box>
  )
}
