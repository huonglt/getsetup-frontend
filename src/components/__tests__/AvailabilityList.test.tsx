import { AvailabilityList } from '../AvailabilityList'
import { render, screen } from '@testing-library/react'
import React from 'react'

const guideAvailability = {
  userId: 1,
  weekNumber: 1,
  availability: [
    {
      from: new Date(2022, 10, 16, 8, 0, 0), // 16 Nov 2022 8am
      to: new Date(2022, 10, 16, 9, 0, 0) // 16 Nov 2022 9am
    }
  ]
}
describe('AvailabilityList', () => {
  it('match snapshot', () => {
    const { container } = render(
      <AvailabilityList guideAvailability={guideAvailability} />
    )
    expect(container).toMatchSnapshot()
  })

  it('render correctly', () => {
    render(<AvailabilityList guideAvailability={guideAvailability} />)
    expect(
      screen.getByText('Teaching Availability for userId 1 for the week #1')
    ).toBeInTheDocument()
    expect(screen.getByText('Wed Nov 16 2022 8:00 am - 9:00 am'))
  })
})
