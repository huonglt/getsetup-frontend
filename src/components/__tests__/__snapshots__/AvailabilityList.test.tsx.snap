// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailabilityList match snapshot 1`] = `
<div>
  <div
    class="spaceTop MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Teaching Availability for userId 
      1
       for the week #
      1
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Wed Nov 16 2022 8:00 am - 9:00 am
    </p>
  </div>
</div>
`;
